config/src/lib.rs

```diff
@@ -1187,39 +1187,205 @@ impl Config {
+                    config_warn!(
+                        "Could not merge [{}] for TOML at {}: [{}.{}] already exists",
+                        profile_str,
+                        toml_path.display(),
+                        Config::PROFILE_SECTION,
+                        profile_str
+                    );
+                    return Some(value)
+                }
+            }
+            // insert the profile
+            profile_map.as_table_like_mut().unwrap().insert(profile_str, value);
+            None
+        };
+        // move [<profile>] to [profile.<profile>]
+        let mut edited = false;
+        for profile in profiles {
+            let profile_str = profile.to_string().to_snake_case();
+            let toml_map = toml_doc.as_table_mut();
+            if let Some(value) = toml_map.remove(&profile_str) {
+                if !value.is_table_like() {
+                    config_warn!(
+                        "Invalid profile [{}] for TOML at {}: Expected [{}] to be a Table",
+                        profile_str,
+                        toml_path.display(),
+                        profile_str
+                    );
+                    toml_map.insert(&profile_str, value);
+                } else if let Some(non_inserted) = insert_profile(toml_map, &profile_str, value) {
+                    toml_map.insert(&profile_str, non_inserted);
+                } else {
+                    edited = true;
+                }
+            } else if profile == Config::DEFAULT_PROFILE &&
+                toml_map
+                    .get(Config::PROFILE_SECTION)
+                    .and_then(|val| val.as_table_like())
+                    .and_then(|table| table.get(&profile_str))
+                    .is_none() &&
+                insert_profile(
+                    toml_map,
+                    &profile_str,
+                    toml_edit::Item::Table(toml_edit::Table::new()),
+                )
+                .is_none()
+            {
+                // inserted default profile if it doesn't exist
+                edited = true;
+            }
+        }
+        // emit warnings for unknown sections
+        for section in
+            toml_doc.as_table().iter().map(|(k, _)| k).filter(|k| {
+                !(k == &Config::PROFILE_SECTION || Config::STANDALONE_SECTIONS.contains(k))
+            })
+        {
+            config_warn!("Unknown section [{}] found in TOML at {}. If you wish to use this section as a profile, run the command again with FOUNDRY_PROFILE=\"{}\".", section, toml_path.display(), section);
+        }
+        // write toml back to file
+        if edited {
+            if let Err(err) = fs::write(toml_path, toml_doc.to_string()) {
+                config_warn!("Could not write TOML to {}: {}", toml_path.display(), err);
+            }
+        }
+    }
+
+    /// Fix foundry.toml files. Making sure any implicit profile [<profile>] becomes
+    /// [profile.<profile>]
+    pub fn fix_tomls() {
```

Comment:
perhaps we'll need more fix functions in the future, since these are stand-alone, can we move them to a `fix.rs` module?